lib/src/client/dio_interceptor.dart

```diff
@@ -0,0 +1,112 @@
+import 'package:dio/dio.dart';
+import 'package:requests_signature_dart/requests_signature_dart.dart';
+
+import 'dart:convert';
+import 'package:requests_signature_dart/src/client/requests_signature_options.dart';
+
+import 'dart:typed_data';
+
+import 'package:uuid/uuid.dart';
+
+/// Interceptor for signing outgoing requests with request signature.
+///
+/// This interceptor signs the outgoing requests with a request signature
+/// before forwarding them to the inner Dio client for processing.
+class RequestsSignatureInterceptor extends Interceptor {
+  Uuid _uuid = const Uuid();
+  final RequestsSignatureOptions _options;
+  final ISignatureBodySourceBuilder _signatureBodySourceBuilder;
+  final ISignatureBodySigner _signatureBodySigner;
+  late int _clockSkew;
+
+  /// Constructs a new [RequestsSignatureInterceptor].
+  ///
+  /// The [options] parameter specifies the signature options.
+  ///
+  /// Optionally, you can provide custom implementations for
+  /// [signatureBodySourceBuilder] and [signatureBodySigner].
+  RequestsSignatureInterceptor(
+    this._options, {
+    ISignatureBodySourceBuilder? signatureBodySourceBuilder,
+    ISignatureBodySigner? signatureBodySigner,
+  })  : _signatureBodySourceBuilder =
+            signatureBodySourceBuilder ?? SignatureBodySourceBuilder(),
+        _signatureBodySigner =
+            signatureBodySigner ?? HashAlgorithmSignatureBodySigner() {
+    _clockSkew = 0;
```

Comment:
This is currently always 0, which isn't equivalent to the .NET implementation.

Reply:
fixed